com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/security/RateLimiterFilter.java

```diff
@@ -0,0 +1,44 @@
+package com.sap.cloud.lm.sl.cf.web.security;
+
+import com.sap.cloud.lm.sl.cf.web.helpers.RateLimiterProvider;
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.internal.AtomicRateLimiter.AtomicRateLimiterMetrics;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.inject.Inject;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+public class RateLimiterFilter extends OncePerRequestFilter {
+
+    @Inject
+    private RateLimiterProvider rateLimiterProvider;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String ipAddress = request.getRemoteAddr();
+        RateLimiter rateLimiter = rateLimiterProvider.getRateLimiter(ipAddress);
+
+        boolean canProcessRequest = rateLimiter.acquirePermission();
+        AtomicRateLimiterMetrics metrics = (AtomicRateLimiterMetrics) rateLimiter.getMetrics();
+
+        if (canProcessRequest) {
+            response.setHeader("X-Rate-Limit-Remaining", Integer.toString(metrics.getAvailablePermissions()));
```

Comment:
Can we be consistent with CF and return the same information they also do?
CloudFoundry returns the following information about rate limiting:
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 56
X-RateLimit-Reset: 1372700873

Take a look @ [Rate Limit info](https://docs.cloudfoundry.org/running/rate-limit-cloud-controller-api.html)

Reply:
Also extract to constants